service/controller/resource/spark/create.go

```diff
@@ -100,6 +100,14 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      secretName(sparkCR.Name),
 					Namespace: azureMachinePool.Namespace,
+					OwnerReferences: []metav1.OwnerReference{
```

Comment:
I set the OwnerReference field in the Secret created to hold the cloudconfig string, that way when the Spark CR is deleted, the Secret will be deleted by Kubernetes because it's an orphan.

Is this the right approach or should I explicitly delete the secret from here?

Reply:
In my view, these are effectively the same. We can delete dependent object in our own finalizer step or by leveraging built-in kubernetes garbage collection via `OwnerReferences`. It means less code for us so it seems preferable to use `OwnerReferences` when possible.